### PR TITLE
Bulk search emails and phone numbers

### PR DESCRIPTION
**Pull request by zapu**
caller also gets assertions as a side effect.

To have this done and also clean `usersearch.go` a little, I extracted the searching machinery previously found in `imptofuSearch` to `searchEmailsOrPhoneNumbers`. This new function can search for multiple phone numbers or email addresses using one provider call (so one API call in the end as well). 

`searchEmailsOrPhoneNumbers` has to do the following:
- make assertions from inputs,
- user contactsProvider to lookup inputs (contactsProvider handles API call, cache tokens etc.),
- use contactsProvider to find usernames and service maps (optionally; contactsProvider just uses `uidmap` package),
- combine all of the above to result list.

`imptofuSearch` had to do similar amount of work but its purpose was very specific. `imptofuSearch` was used for the GUI phone / email tabs so it could only search for just number or email, and also it was tied to `ImpTofuSearchResult` record.

`imptofuSearch` now calls `searchEmailsOrPhoneNumbers` with one of either emails or phone numbers, and expects one result.

### CODE REVIEW

**Comment by zapu** [comment]
cc @jakob223

**Comment by zapu** [comment]
Let me know if that's what GUI needs. If not I can work on this more, it's rather easy to mold it to whatever we need because the underlying search function is fairly flexible.

File: go/service/usersearch.go [hunk 1]
```diff
@@ -297,6 +298,146 @@ func imptofuQueryToAssertion(ctx context.Context, typ, val string) (ret libkb.As
...
+			emailRes[i].assertion = assertion
+			emailsToSearch = append(emailsToSearch, email)
+		} else {
+			mctx.Debug("Failed to create assertion from email: %s, skipping in search", email)
+		}
+	}
+
+	phonesToSearch := make([]keybase1.RawPhoneNumber, 0, len(phoneNumbers))
+	phoneRes := make([]imptofuQueryResult, len(phoneNumbers))
+	for i, phone := range phoneNumbers {
```
Comment (heronhaye) [inline go/service/usersearch.go]: tfw no generics

File: go/service/usersearch.go [hunk 2]
```diff
@@ -574,3 +670,79 @@ func (h *UserSearchHandler) GetNonUserDetails(ctx context.Context, arg keybase1.
 
 	return res, nil
 }
+
+// 🍝 splitBulk splits on newline or comma.
```
Comment (heronhaye) [inline go/service/usersearch.go]: no emoji please

File: go/service/usersearch.go [hunk 3]
```diff
@@ -574,3 +670,79 @@ func (h *UserSearchHandler) GetNonUserDetails(ctx context.Context, arg keybase1.
...
+	mctx := libkb.NewMetaContext(ctx, h.G())
+	defer mctx.TraceTimed(fmt.Sprintf("UserSearch#BulkEmailOrPhoneSearch(%d emails,%d phones)",
+		len(arg.Emails), len(arg.PhoneNumbers)), func() error { return err })()
+
+	splitEmailCandidates := splitBulk(arg.Emails)
+	emails := make([]keybase1.EmailAddress, 0, len(splitEmailCandidates))
+	for _, maybeEmail := range splitEmailCandidates {
+		address, err := mail.ParseAddress(maybeEmail)
+		if err != nil {
+			mctx.Debug("Failed to mail.ParseAddress(%q): %s", maybeEmail, address)
```
Comment (heronhaye) [inline go/service/usersearch.go]: should include error?

**Comment by jakob223** [comment]
I think this definitely does what the GUI needs, thanks @zapu

**Pull request merged**